### PR TITLE
feat: 327 mobility database change gtfs components to gtfs features label

### LINKED ISSUE #327: Mobility Database: Change GTFS components to GTFS features label
Tasks: 
 - [x] Concurrent changes in the Mobility Database API schema.
 - [x] Change API DB schema. Review populate script.
 - [x] Change API report endpoint implementation.
 - [x] Update documentation to mobilitydata catalog.

### PR DESCRIPTION
**Pull request by qcdyx**
**Summary:**

Closes #327 
updated GTFS Components to GTFS Features label
image
image

**Expected behavior:** 
No GTFS Components found in the Mobility Database API, components are updated to features.

**Testing tips:**

Provide tips, procedures and sample files on how to test the feature.
Testers are invited to follow the tips AND to try anything they deem relevant outside the bounds of the testing tips. 

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `./scripts/api-tests.sh` to make sure you didn't break anything
- [ ] Add or update any needed documentation to the repo
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [x] Linked all relevant issues
- [x] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

**Comment by emmambd** [comment]
@qcdyx Will there be an associated PR for the schema changes on the catalogs repo too?

File: api/src/feeds/impl/datasets_api_impl.py [hunk 1]
```diff
@@ -40,29 +40,18 @@ class DatasetsApiImpl(BaseDatasetsApi):
...
+        return Query(
+            [
+                Gtfsdataset,
+                Gtfsdataset.bounding_box.ST_AsGeoJSON(),
+                Feed.stable_id,
+            ]
+        ).join(Feed, Feed.id == Gtfsdataset.feed_id)
 
     @staticmethod
     def _load_validation_report():
```
Comment (cka-y) [inline api/src/feeds/impl/datasets_api_impl.py:52]: I've noticed that this method is responsible for loading the notices. Given that we work with historical datasets, which include all report versions for all datasets, the volume of data retrieved could become significantly large. To optimize our data handling and focus on relevant information, we  might want to consider implementing a filter by `dataset_id`? This way, we can streamline the process by directly accessing the data that's most pertinent.

File: liquibase/changelog.xml [hunk 2]
```diff
@@ -18,4 +18,5 @@
     <include file="changes/feat_240.sql" relativeToChangelogFile="true"/>
     <include file="changes/feat_263.sql" relativeToChangelogFile="true"/>
      <include file="changes/feat_66.sql" relativeToChangelogFile="true"/>
```
Comment (cka-y) [inline liquibase/changelog.xml:20]:
```suggestion
    <include file="changes/feat_66.sql" relativeToChangelogFile="true"/>
```

File: liquibase/changes/feat_327.sql [hunk 3]
```diff
@@ -0,0 +1,9 @@
+ALTER TABLE component
+  RENAME TO feature;
+DROP TABLE componentgtfsdataset;
+
```
Comment (cka-y) [inline liquibase/changes/feat_327.sql:4]: Suggestion for easier reading:
```suggestion
ALTER TABLE Component
  RENAME TO Feature;
DROP TABLE ComponentGtfsDataset;

```

File: api/src/feeds/impl/datasets_api_impl.py (hunk 1 above)
Comment (qcdyx) [reply api/src/feeds/impl/datasets_api_impl.py:52]: This is a good point. Shall we create a new story for data retrieval optimization? @emmambd @davidgamez

**Comment by qcdyx** [comment]
> @qcdyx Will there be an associated PR for the schema changes on the catalogs repo too?

Please correct me if I'm wrong, but I don't find usages of "component" in the database catalog repo except the following code block, which is something different than GTFS components. @emmambd 

```
    var csvLines : [String] = csvData.components(separatedBy: csvLineSeparator) ; csvLines.removeFirst(1) ; var csvArray : [[String]] = []
    for currentLine : String in csvLines {
        if currentLine.count > 5 { csvArray.append(currentLine.components(separatedBy: csvColumnSeparator)) }
    }
```

**Comment by emmambd** [comment]
@qcdyx The main place is in the Python operations, I think, and the schema itself. We could postpone that change though until we modify the sources.csv and make a 2.0 version though...so let's keep it as is.

**Review by cka-y — APPROVED** [review]
LGTM

**Pull request merged**